compiler/rustc_codegen_ssa/src/back/metadata.rs

```diff
@@ -93,6 +94,56 @@ pub(super) fn search_for_section<'a>(
         .map_err(|e| format!("failed to read {} section in '{}': {}", section, path.display(), e))
 }
 
+fn add_gnu_property_note(
+    sess: &Session,
+    file: &mut write::Object<'static>,
+    architecture: Architecture,
+    binary_format: BinaryFormat,
+) {
+    // check bti protection
+    let mut check_cfprotection = false;
+    let mut pr_type: u32 = 0;
+    if architecture == Architecture::X86_64
+        && matches!(
+            sess.opts.unstable_opts.cf_protection,
+            rustc_session::config::CFProtection::Branch | rustc_session::config::CFProtection::Full
+        )
+    {
+        check_cfprotection = true;
+        pr_type = 0xc0000002;
+    }
+    if architecture == Architecture::Aarch64 && sess.opts.unstable_opts.branch_protection.is_some()
+    {
+        check_cfprotection = true;
+        pr_type = 0xc0000000;
+    }
+    if check_cfprotection && binary_format == BinaryFormat::Elf {
+        let segment: Vec<u8> = file.segment_name(StandardSegment::Data).to_vec();
+        let name: Vec<u8> = b".note.gnu.property".to_vec();
+        let kind = SectionKind::Note;
+        let section = file.add_section(segment, name, kind);
+        let mut data: Vec<u8> = Vec::new();
+        let n_namsz: u32 = 4; // Size of the n_name field
+        let n_descsz: u32 = 16; // Size of the n_desc field
+        let n_type: u32 = NT_GNU_PROPERTY_TYPE_0; // Type of note descriptor
+        let values = [n_namsz, n_descsz, n_type];
+        values.map(|v| data.extend_from_slice(&(v.to_ne_bytes())));
```

Comment:
Is native byte ordering right here? If we're cross compiling, I don't think the result should depend on the **host** endianess.

Reply:
I changed it to little-endian. Thanks for your comment.